Fix shift overflow bug in preincrement and predecrement operators

At the moment the preincrement and predecrement operations are creating the following warnings:
> SFixedMemberFunctions.h:224:20: warning: left shift count >= width of type [-Wshift-count-overflow]
>
>  this->value += (1 << FractionSize);

> SFixedMemberFunctions.h:231:20: warning: left shift count >= width of type [-Wshift-count-overflow]
> 
>   this->value -= (1 << FractionSize);

In the code generating these warnings I am using `SQ15x16` and targetting an AVR architecture (Arduino Uno R3) so the likely cause is that `1` is interpreted as being an `int`, which is 16 bits wide on this architecture, and `FractionSize` has a value of `16` for `SQ15x16` which means the integer is being shifted past its full width.

The optimal solution is probably to cast the `1` to `InternalType` before the shift.

--- a/src/FixedPoints/SFixedMemberFunctions.h
+++ b/src/FixedPoints/SFixedMemberFunctions.h
@@ -221,14 +221,14 @@ constexpr SFixed<Integer, Fraction> SFixed<Integer, Fraction>::operator -() cons
 template< unsigned Integer, unsigned Fraction >
 SFixed<Integer, Fraction> & SFixed<Integer, Fraction>::operator ++()
 {
-	this->value += (1 << FractionSize);
+	this->value += (static_cast<InternalType>(1) << FractionSize);
 	return *this;
 }
 
 template< unsigned Integer, unsigned Fraction >
 SFixed<Integer, Fraction> & SFixed<Integer, Fraction>::operator --()
 {
-	this->value -= (1 << FractionSize);
+	this->value -= (static_cast<InternalType>(1) << FractionSize);
 	return *this;
 }
 

--- a/src/FixedPoints/UFixedMemberFunctions.h
+++ b/src/FixedPoints/UFixedMemberFunctions.h
@@ -206,14 +206,14 @@ constexpr UFixed<Integer, Fraction> UFixed<Integer, Fraction>::fromInternal(cons
 template< unsigned Integer, unsigned Fraction >
 UFixed<Integer, Fraction> & UFixed<Integer, Fraction>::operator ++()
 {
-	this->value += (1 << FractionSize);
+	this->value += (static_cast<InternalType>(1) << FractionSize);
 	return *this;
 }
 
 template< unsigned Integer, unsigned Fraction >
 UFixed<Integer, Fraction> & UFixed<Integer, Fraction>::operator --()
 {
-	this->value -= (1 << FractionSize);
+	this->value -= (static_cast<InternalType>(1) << FractionSize);
 	return *this;
 }
 

